pkg/controller/resourceclaim/controller_test.go

```diff
@@ -511,6 +509,96 @@ func TestSyncHandler(t *testing.T) {
 	}
 }
 
+func TestResourceClaimEventHandler(t *testing.T) {
+	tCtx := ktesting.Init(t)
+	tCtx = ktesting.WithCancel(tCtx)
+
+	fakeKubeClient := createTestClient()
+	setupMetrics()
+	informerFactory := informers.NewSharedInformerFactory(fakeKubeClient, controller.NoResyncPeriodFunc())
+	podInformer := informerFactory.Core().V1().Pods()
+	podSchedulingInformer := informerFactory.Resource().V1alpha3().PodSchedulingContexts()
+	claimInformer := informerFactory.Resource().V1alpha3().ResourceClaims()
+	templateInformer := informerFactory.Resource().V1alpha3().ResourceClaimTemplates()
+	claimClient := fakeKubeClient.ResourceV1alpha3().ResourceClaims(testNamespace)
+
+	_, err := NewController(tCtx.Logger(), fakeKubeClient, podInformer, podSchedulingInformer, claimInformer, templateInformer)
+	tCtx.ExpectNoError(err, "creating ephemeral controller")
+
+	informerFactory.Start(tCtx.Done())
```

Comment:
Do we need to add `informerFactory.WaitForCacheSync` ?

Reply:
No. The fake client has no objects, so there is nothing that needs to be synced into the cache.